ibm/service/power/data_source_ibm_pi_network_security_group.go

```diff
@@ -0,0 +1,320 @@
+				ValidateFunc: validation.NoZeroValues,
+			},
+			// Attributes
+			Attr_CRN: {
+				Computed:    true,
+				Description: "The network security group's crn.",
+				Type:        schema.TypeString,
+			},
+			Attr_Members: {
+				Computed:    true,
+				Description: "The list of IPv4 addresses and, or network interfaces in the network security group.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						Attr_ID: {
+							Computed:    true,
+							Description: "The ID of the member in a network security group.",
+							Type:        schema.TypeString,
+						},
+						Attr_MacAddress: {
+							Computed:    true,
+							Description: "The mac address of a network interface included if the type is network-interface.",
+							Type:        schema.TypeString,
+						},
+						Attr_Target: {
+							Computed:    true,
+							Description: "If ipv4-address type, then IPv4 address or if network-interface type, then network interface ID.",
+							Type:        schema.TypeString,
+						},
+						Attr_Type: {
+							Computed:    true,
+							Description: "The type of member.",
+							Type:        schema.TypeString,
+						},
+					},
+				},
+				Type: schema.TypeList,
+			},
+			Attr_Name: {
+				Computed:    true,
+				Description: "The name of the network security group.",
+				Type:        schema.TypeString,
+			},
+			Attr_Rules: {
+				Computed:    true,
+				Description: "The list of rules in the network security group.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						Attr_Action: {
+							Computed:    true,
+							Description: "The action to take if the rule matches network traffic.",
+							Type:        schema.TypeString,
+						},
+						Attr_DestinationPort: {
+							Computed:    true,
+							Description: "The list of destination port.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									Attr_Maximum: {
+										Computed:    true,
+										Description: "The end of the port range, if applicable, If values are not present then all ports are in the range.",
+										Type:        schema.TypeFloat,
+									},
+									Attr_Minimum: {
+										Computed:    true,
+										Description: "The start of the port range, if applicable. If values are not present then all ports are in the range.",
+										Type:        schema.TypeFloat,
+									},
+								},
+							},
+							Type: schema.TypeList,
+						},
+						Attr_ID: {
+							Computed:    true,
+							Description: "The ID of the rule in a network security group.",
+							Type:        schema.TypeString,
+						},
+						Attr_Name: {
+							Computed:    true,
+							Description: "The unique name of the network security group rule.",
+							Type:        schema.TypeString,
+						},
+						Attr_Protocol: {
+							Computed:    true,
+							Description: "The list of protocol.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									Attr_ICMTypes: {
```

Comment:
```suggestion
									Attr_ICPMTypes: {
```

Reply:
@ismirlia what is the suggestion change?

Reply:
the field is Attr_ICMPTypes not Attr_ICMTypes right?

Reply:
I was reading it as the same :joy: and couldn't tell the difference.